jdbc/src/main/java/org/apache/zeppelin/jdbc/JDBCInterpreter.java

```diff
@@ -421,13 +411,31 @@ public Connection run() throws Exception {
             break;
 
           default:
-            connection = getConnectionFromPool(url, user, propertyKey, properties);
+            StringBuilder connectionUrl = new StringBuilder(url);
+            checkAndAppendHiveProxyUser(connectionUrl, user);
+            connection = getConnectionFromPool(connectionUrl.toString(),
+                user, propertyKey, properties);
       }
     }
     propertyKeySqlCompleterMap.put(propertyKey, createSqlCompleter(connection));
     return connection;
   }
 
+  private void checkAndAppendHiveProxyUser(StringBuilder connectionUrl, String user) {
+    if (connectionUrl.toString().trim().startsWith("jdbc:hive")) {
```

Comment:
I thought about other approaches, like using `propertyKey`, but even in that case name could be anything including default.